For BYO Tool use case with Nock (record/replay example), user should be reminded to set PACT_PROVIDER env var

Env var not required for non BYO case, as it has a fallback. 

Possibly add the fallback, or add into the instructions, reminder to set PACT_PROVIDER env var.

```
{"errors":{"contracts":["providerName is missing at index 0"]}}
make[1]: *** [publish_pacts] Error 1
make: *** [fake_ci_nock] Error 2
```

api.pact.spec.js
`  provider: process.env.PACT_PROVIDER ? process.env.PACT_PROVIDER : 'pactflow-example-provider',`

api.record.spec.js
` provider: { name: process.env.PACT_PROVIDER },`

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -14,17 +14,24 @@ env:
 jobs:
   test:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        pact_provider: ['pactflow-example-provider-postman', 'pactflow-example-provider']
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-node@v2-beta
         with:
           node-version: '12'
-      - name: Install
+      - name: Install for ${{ matrix.pact_provider }}
         run: npm i
       - name: Test
         run: make test
-      - name: Publish pacts
+        env:
+          PACT_PROVIDER: ${{ matrix.pact_provider }}
+      - name: Publish pacts for ${{ matrix.pact_provider }}
         run: GIT_BRANCH=${GIT_REF:11} make publish_pacts
+        env:
+          PACT_PROVIDER: ${{ matrix.pact_provider }}
 
   # Runs on branches as well, so we know the status of our PRs
   can-i-deploy:

--- a/src/api.record.spec.js
+++ b/src/api.record.spec.js
@@ -30,7 +30,7 @@ export const convertNockToPact = () => {
 
   const pact = {
     consumer: { name: "pactflow-example-consumer" },
-    provider: { name: process.env.PACT_PROVIDER },
+    provider: { name: process.env.PACT_PROVIDER ? process.env.PACT_PROVIDER : 'pactflow-example-provider' },
     interactions: [],
     metadata: {
       pactSpecification: {

